### ISSUE TITLE
"Run Code" should be implemented based on DAP (Debug Adapter Protocol)

### ISSUE THREAD

**Issue by tokoro10g** [tokoro10g/matlab-langserver#4]
Details:
https://microsoft.github.io/debug-adapter-protocol/

lsp4j supports DAP:
https://github.com/eclipse/lsp4j

**Comment by minkir014** [comment]
You can use `gdb` as a debugger and do like Native Debug extension does

https://marketplace.visualstudio.com/items?itemName=webfreak.debug

**Comment by tokoro10g** [comment]
@minkir014 
No, because MATLAB script can only be executed on MATLAB engine, your suggestion does not apply.
The implementation might be a useful reference though.

**Comment by minkir014** [comment]
You have two ways for doing that first you could compile matlab to c files and debug it with gdb see [here](https://blogs.mathworks.com/developer/2018/06/19/mex-debugging-vscode/).


You have another solution and it's the hardest that you should debug matlab over the matlab interactive terminal by sending commands to tell it where to stop and if to continue or to step into function. Tell me which one do you prefer??

**Comment by minkir014** [comment]
@tokoro10g, Any new news? Have you decided which way of the ways I have mentioned you want to implement????

**Comment by tokoro10g** [comment]
I have looked at DAP specifications, but it seems not suitable for MATLAB
debugging. This is because the common way in MATLAB is out of the scope of
DAP.

For example, we often use "Run section", "Run selection", and interactive
command window, but DAP only supports
executing the entire code.
There are several more issues which makes me unwilling to adapt DAP.

Instead, I found an interesting project which aims at providing an
alternative to MATLAB IDE using VS Code.
https://github.com/marmottetranquille/aMi
This is directly implemented on the extension API. It might not be
compatible with neither DAP nor LSP, but I think this works for you.

BTW, please do not push me since this project is a kind of hobby hack, and
you cannot consider this as a fully-supported software.
